evm/src/main/java/org/hyperledger/besu/evm/gascalculator/OsakaGasCalculator.java

```diff
@@ -99,19 +99,18 @@ public long modExpGasCost(final Bytes input) {
         clampedAdd(BigIntegerModularExponentiationPrecompiledContract.BASE_OFFSET, baseLength);
 
     final long maxLength = Math.max(modulusLength, baseLength);
-    if (maxLength <= 0) {
-      return 500L;
-    }
```

Comment:
think it would still be better performance to return early if values are zero

Reply:
But that is incorrect. That is what the PR is fixing

Reply:
ah 😅 I see